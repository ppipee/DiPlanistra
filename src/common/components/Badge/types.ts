export enum BadgeVariant {
	Outlined = 'outlined',
	Filled = 'filled',
}

export type BadgeVariantTypes = BadgeVariant | 'filled' | 'outlined'

export interface BadgeProps {
	$variant?: BadgeVariantTypes
	$color: string
	$secondColor?: string
}
