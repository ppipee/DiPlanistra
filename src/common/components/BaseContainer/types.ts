import { ScreenName } from 'common/styles/utils/viewport/screenSizes'

export interface BaseContainerProps {
	$screen?: ScreenName
	$padding?: string
	$paddingMobile?: string
	$margin?: string
	$marginMobile?: string
}
