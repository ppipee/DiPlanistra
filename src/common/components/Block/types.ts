export interface BlockProps {
	$padding?: string
	$borderRadius?: string
	$margin?: string
}
