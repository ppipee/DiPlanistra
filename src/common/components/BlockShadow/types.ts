export enum Shadow {
	Default = 'default',
	Hard = 'hard',
}

type ShadowType = Shadow | Shadow.Default | Shadow.Hard

export interface BlockShadowProps {
	$variant?: ShadowType
}
