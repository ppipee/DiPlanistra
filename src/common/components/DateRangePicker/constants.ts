// CUSTOM_INDEX should more or less than range of PICKER_CONTROLLERS
export const PICKER_CUSTOM_INDEX = -1
