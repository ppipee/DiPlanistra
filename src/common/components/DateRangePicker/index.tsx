import React, { ReactNode, useCallback } from 'react'

import { CalendarCellDate } from 'common/components/Calendar/components/CalendarCell'
import { CalendarDisplayRange, CalenderDateProps } from 'common/components/Calendar/types'
import DateRangeProvider from 'common/components/DateRangePicker/components/DateRangeProvider'

import RangePickerPanel from './components/RangePickerPanel'

type Props = {
	date: CalenderDateProps
	setDate: (date: CalenderDateProps) => void
	dateRangeShown?: CalendarDisplayRange
	children?: ReactNode
	isDateSelectable?: (date: CalendarCellDate) => boolean
	showHeader?: boolean
}

const DateRangePicker = ({ date, setDate, isDateSelectable, dateRangeShown, children, showHeader }: Props) => {
	const onSetDate = useCallback(({ endDate, startDate }: CalenderDateProps = {}) => {
		setDate({ endDate, startDate })
	}, [])

	return (
		<DateRangeProvider
			isDateSelectable={isDateSelectable}
			date={date}
			setDate={onSetDate}
			dateRangeShown={dateRangeShown}
		>
			<RangePickerPanel showHeader={showHeader}>{children}</RangePickerPanel>
		</DateRangeProvider>
	)
}

export default DateRangePicker
