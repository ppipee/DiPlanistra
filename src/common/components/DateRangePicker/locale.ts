export const PICKER_YESTERDAY = ['เมื่อวาน', 'Yesterday']
export const PICKER_LAST_7_DAYS = ['7 วันที่แล้ว', 'Last 7 Days']
export const PICKER_LAST_14_DAYS = ['14 วันที่แล้ว', 'Last 14 Days']
export const PICKER_LAST_30_DAYS = ['30 วันที่แล้ว', 'Last 30 Days']
export const PICKER_THIS_MONTH = ['เดือนนี้', 'This Month']
export const CUSTOM_PICKER = ['กำหนดเอง', 'Custom']
