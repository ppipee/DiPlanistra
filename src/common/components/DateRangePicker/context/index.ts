import { createContext, useContext } from 'react'

import { CalendarCellDate } from 'common/components/Calendar/components/CalendarCell'
import { CalendarDisplayRange, CalenderDateProps } from 'common/components/Calendar/types'

export interface DateRangeContextType {
	date: CalenderDateProps
	setDate: (date: CalenderDateProps) => void
	controllerIndex: number | null
	selectController: (state: number | null) => void
	isDateSelectable: (date: CalendarCellDate) => boolean
	dateRangeShown?: CalendarDisplayRange
}

const DateRangeContext = createContext<DateRangeContextType | null>(null)

export const useDateRangeContext = () => useContext(DateRangeContext)!

export default DateRangeContext
