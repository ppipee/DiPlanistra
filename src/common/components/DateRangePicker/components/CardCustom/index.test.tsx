import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import { createI18nWrapper } from 'core/locale/testUtils'
import { Locale } from 'core/locale/types'

import { PICKER_CUSTOM_INDEX } from 'common/components/DateRangePicker/constants'

import * as CardCustomType from '.'

describe('Test CardCustom', () => {
	const useDateRangeContextSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/context', () => ({
		useDateRangeContext: useDateRangeContextSpy,
	}))

	const { default: CardCustom } = require('.') as typeof CardCustomType

	afterEach(() => {
		jest.resetAllMocks()
	})

	beforeEach(() => {
		useDateRangeContextSpy.mockReturnValue({
			date: {},
			isSelected: PICKER_CUSTOM_INDEX,
		})
	})

	it('should render CardCustom with TH text correctly', () => {
		const { getByText } = render(<CardCustom />, {
			wrapper: createI18nWrapper(Locale.th),
		})

		expect(getByText('กำหนดเอง')).toBeDefined()
	})

	it('should render CardCustom with EN text correctly', () => {
		const { getByText } = render(<CardCustom />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(getByText('Custom')).toBeDefined()
	})

	it('should render with active styled', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: {
				startDate: new Date(2020, 5, 1),
			},
			isSelected: PICKER_CUSTOM_INDEX,
		})

		const { container } = render(<CardCustom />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render without active styled', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: {
				startDate: new Date(2020, 5, 1),
			},
			isSelected: PICKER_CUSTOM_INDEX,
		})

		const { container } = render(<CardCustom />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(container.firstChild).toMatchSnapshot()
	})
})
