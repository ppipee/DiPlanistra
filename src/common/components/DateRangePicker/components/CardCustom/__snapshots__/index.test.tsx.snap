// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardCustom should render with active styled 1`] = `
.c0 {
  width: 140px;
  height: 40px;
  padding: 8px;
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 4px;
  background-color: #E9EAEC;
  color: #7D8391;
}

.c1 {
  cursor: default;
}

<div
  class="c0 c1 heading4"
>
  Custom
</div>
`;

exports[`Test CardCustom should render without active styled 1`] = `
.c0 {
  width: 140px;
  height: 40px;
  padding: 8px;
  box-sizing: border-box;
  cursor: pointer;
  border-radius: 4px;
  background-color: #E9EAEC;
  color: #7D8391;
}

.c1 {
  cursor: default;
}

<div
  class="c0 c1 heading4"
>
  Custom
</div>
`;
